### ISSUE TITLE
File.OpenFolderDialog not waiting, caller continues before dialog opens

### ISSUE THREAD

**Issue by sjsepan3** [VBAndCs/sVB-Small-Visual-Basic#14]
GetFolderName appears to return with empty value before dialog is shown; the value I select in the dialog is not getting into the variable I check.

`
'------------------------------------------------
Function GetFolderName()
   returnValue = ""
   
   WriteError("before")
   returnValue = File.OpenFolderDialog("c:\")
   WriteError("after")
   Return returnValue
EndFunction

`

**Comment by VBAndCs** [comment]
@sjsepan3 
I mistakenly called the SmallBasicApplication.BeginInvoke method at the start of the (OpenFolderDialog function]( https://github.com/VBAndCs/sVB-Small-Visual-Basic/blob/c0907f804064d810d6eb25b8b99667c4c5137b1a/SmallBasicLibrary/Library/File.vb#L690), which invokes the given method after the currnet method is completed. I just should call SmallBasicApplication.Invoke (which I use in all other places in the lib).
Also, the `Return ""` statement at the end of the function must be removed.
I fixed the bug and the fix will be available in the next update.
Thanks.

**Comment by VBAndCs** [comment]
@sjsepan3
It's available now in [sVB v2.8.4](https://marketplace.visualstudio.com/manage/publishers/modernvbnet?src=ModernVBNET.sVBInstaller). In this version you can also use the turtle can fill an area, by calling the CreateFigure and FillFigure methods.
```vb.net
X = Turtle.X
Y = Turtle.Y

Turtle.CreateFigure()
Turtle.Move(200)
Turtle.TurnRight()
Turtle.Move(200)
Turtle.TurnRight()
Turtle.Move(200)
Turtle.TurnRight()
Turtle.Move(200)
Turtle.FillFigure()

Turtle.CreateFigure()
Turtle.TurnLeft()
Turtle.Move(200)
Turtle.MoveTo(200, 200)
Turtle.MoveTo(350, 200)
Turtle.MoveTo(X, Y)
GraphicsWindow.BrushColor = Colors.Red
Turtle.FillFigure()
```

![image](https://github.com/VBAndCs/sVB-Small-Visual-Basic/assets/48354902/42b207a9-1038-4f05-b206-792fffef3257)

**Comment by sjsepan3** [comment]
File.OpenFolderDialog works great here too. Thanks!
